folium/features.py

```diff
@@ -610,3 +585,60 @@ def __init__(self, image, bounds, opacity=1., attribution=None, origin='upper',
                     ).addTo({{this._parent.get_name()}});
             {% endmacro %}
             """)
+
+class CustomIcon(Icon):
```

Comment:
I feel so ashamed for wasting an afternoon doing [this](https://github.com/ocefpaf/secoora_assets_map/blob/gh-pages/assets_map.ipynb) :flushed: 

This class is pure art!